conseil-common/src/main/scala/tech/cryptonomic/conseil/common/rpc/RpcClient.scala

```diff
@@ -0,0 +1,93 @@
+package tech.cryptonomic.conseil.common.rpc
+
+import scala.util.control.NoStackTrace
+
+import cats.effect.{Concurrent, Resource}
+import com.typesafe.scalalogging.LazyLogging
+import fs2.Stream
+import org.http4s.{EntityDecoder, EntityEncoder, Header, Method, Uri}
+import org.http4s.client.Client
+import org.http4s.client.dsl.Http4sClientDsl
+
+import tech.cryptonomic.conseil.common.rpc.RpcClient._
+
+// JSON-RPC client according to specification: https://www.jsonrpc.org/specification
+class RpcClient[F[_]: Concurrent](
+    endpoint: String,
+    maxConcurrent: Int,
+    httpClient: Client[F],
+    headers: Header*
+) extends Http4sClientDsl[F]
+    with LazyLogging {
+
+  def stream[P, R](batchSize: Int)(request: Stream[F, RpcRequest[P]])(
+      implicit decode: EntityDecoder[F, Seq[RpcResponse[R]]],
+      encode: EntityEncoder[F, List[RpcRequest[P]]]
+  ): Stream[F, R] =
+    request
+      .chunkN(batchSize)
+      .evalTap { requests =>
+        Concurrent[F].delay(logger.debug(s"Call RPC in a batch of: ${requests.size}"))
```

Comment:
I've wanted to keep code simple. Should I create a separate effectful logger or just a private method in the client?

Reply:
I was thinking of the second, I understand your objection but my thinking goes:
"Is it simpler to replicate the same verbose and partly non-obvious code everywhere we need to do logging, or is it simpler to write a basic function that does that, with a clear name, that will relieve the burden of reading the call site?"

No obvious answer here, it might be that the function is more complex and less obvious still, so...

Also, consider that any future contributor to the code that will add a log line should see how to do that, and a dedicated function might simplify its job.